csrc/multidevice/utils.cpp

```diff
@@ -168,6 +168,9 @@ bool haveDifferentShardings(TensorView* producer, TensorView* consumer) {
 }
 
 bool isResharding(Expr* expr) {
+  if (!ir_utils::isTvOp(expr)) {
+    return false;
```

Comment:
It's not precisely SliceOp, but the index computation used by Slice. For example: `i13 = fmin(i0, 5)`

This should be correct, because any expression that doesn't have a TV can't be resharding, since it only operates on scalars. 
It looks like almost all expressions with at least one input or output Tv is a TvOp, but there might be a few edge cases since the function checks the exact type of Op. https://github.com/NVIDIA/Fuser/blob/25a3426f1672d1a81db8faf272f5ba18463282db/csrc/device_lower/utils.cpp#L129
From this list I can't tell what ops aren't included, but potentially the ops not in the list would cause an issue...

Reply:
got it, nvm! I had a misconception of what `TvOp` was